xbmc/guilib/GUIControl.cpp

```diff
@@ -124,21 +124,25 @@ void CGUIControl::DynamicResourceAlloc(bool bOnOff)
 // 1. animate and set animation transform
 // 2. if visible, process
 // 3. reset the animation transform
-void CGUIControl::DoProcess(unsigned int currentTime, CDirtyRegionList &dirtyregions)
+void CGUIControl::DoProcess(CGUIRenderInfo &renderInfo)
 {
   CRect dirtyRegion = m_renderRegion;
 
   bool changed = m_bInvalidated && IsVisible();
 
-  changed |= Animate(currentTime);
+  changed |= Animate(renderInfo.GetTime());
 
+  if (m_controlID != WINDOW_FULLSCREEN_VIDEO)
+    renderInfo.IncTotal();
   if (IsVisible())
   {
+    if (m_controlID != WINDOW_FULLSCREEN_VIDEO && !g_windowManager.IsAddonWindow(m_controlID) && !g_windowManager.IsPythonWindow(m_controlID))
+      renderInfo.IncVisible();
```

Comment:
visibility expressions from skin should be checked as well.
IsVisible() only means that its "general visible", the visibility rules can change from true -> false